### PR TITLE
[stable/wordpress] Allow  helm chart to receive an existing database or create it by itself Open, HighAll Users

### PR DESCRIPTION
**Pull request by jbianquetti-nami**
Summary of changes:

    WordPress module does not longer expect the MARIADB_USER input
    WordPress module does not longer expect the MARIADB_PASSWORD input
    WordPress module now expects the WORDPRESS_DATABASE_NAME input
    WordPress module now expects the WORDPRESS_DATABASE_USER input
    WordPress module now expects the WORDPRESS_DATABASE_PASSWORD input

Now, we can also create the database from the application side.
The ENV vars are:

    MYSQL_CLIENT_CREATE_DATABASE_NAME
    MYSQL_CLIENT_CREATE_DATABASE_USER
    MYSQL_CLIENT_CREATE_DATABASE_PASSWORD

Also, we have added an additional security measure regarding passwordless logins. 

`"allowEmptyPassword": {"default": "no", "type": "choice", "validValues": ["yes", "no"], "description": "Allow to set an empty password"}`

In case the container does not have the related ENV var enabled. The ENV var is:

   ` ALLOW_EMPTY_PASSWORD`

### CODE REVIEW

**Comment by jbianquetti-nami** [comment]
CLA signed.

File: stable/wordpress/values.yaml [hunk 1]
```diff
@@ -39,6 +39,28 @@ wordpressLastName: LastName
...
+
+## Database password for the new user
+## Sets WORDPRESS_DATABASE_PASSWORD  and  MYSQL_CLIENT_CREATE_DATABASE_PASSWORD 
+## ref: https://github.com/bitnami/bitnami-docker-wordpress#environment-variables
+## databasePassword: 
+
+## Set to `yes` to allow the container to be started with blank passwords
+## ref: https://github.com/bitnami/bitnami-docker-wordpress#environment-variables
+## Defaults to no
+## allowEmptyPassword:
```
Comment (sameersbn) [inline stable/wordpress/values.yaml]: we generally add the default value to the values.yaml file. I'd suggest un-commenting changing this to

```
allowEmptyPassword: "no"
```

or if you think https://github.com/kubernetes/charts/pull/773/files#r105334999 makes sense, then you should be using

```
allowEmptyPassword: false
```

File: stable/wordpress/templates/deployment.yaml [hunk 2]
```diff
@@ -19,15 +19,35 @@ spec:
         image: "{{ .Values.image }}"
         imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
         env:
+        - name: ALLOW_EMPTY_PASSWORD
+          value: {{ default "no" .Values.allowEmptyPassword | quote }}
```
Comment (sameersbn) [inline stable/wordpress/templates/deployment.yaml]: if user specifies `allowEmptyPassword` value without qoutes, go will treat this as a boolean value. User will HAVE TO specify the `allowEmptyPassword` with double quotes in values.yaml, eg. `allowEmptyPassword: "yes"`. I would not regard this as a stable UX. Instead i suggest using the following snippet. 

```go
        {{- if .Values.allowEmptyPassword }}
        - name: ALLOW_EMPTY_PASSWORD
          value: "yes"
        {{- end }}  
```

or 

```go
        - name: ALLOW_EMPTY_PASSWORD
        {{- if .Values.allowEmptyPassword }}
          value: "yes"
        {{- else }}
          value: "no"
        {{- end }} 
```

File: stable/wordpress/values.yaml [hunk 3]
```diff
@@ -39,6 +39,28 @@ wordpressLastName: LastName
...
+## ref: https://github.com/bitnami/bitnami-docker-wordpress#environment-variables
+## databasePassword: 
+
+## Set to `yes` to allow the container to be started with blank passwords
+## ref: https://github.com/bitnami/bitnami-docker-wordpress#environment-variables
+## Defaults to no
+## allowEmptyPassword: 
+
+
+
```
Comment (sameersbn) [inline stable/wordpress/values.yaml]: consider removing these white spaces

File: stable/wordpress/README.md [hunk 4]
```diff
@@ -55,6 +55,10 @@ The following tables lists the configurable parameters of the WordPress chart an
 | `wordpressFirstName`                 | First name                                 | `FirstName`                                                |
 | `wordpressLastName`                  | Last name                                  | `LastName`                                                 |
 | `wordpressBlogName`                  | Blog name                                  | `User's Blog!`                                             |
+| `databaseName`                       | Database name to create                    | `bitnami_wordpress`                                        |
```
Comment (sameersbn) [inline stable/wordpress/README.md]: The mariadb chart allows you to create a database and user credentials to access the database. Rather than configuring `databaseName`, `databaseUser` and `databasePassword` on the wordpres chart it think `mariadbDatabase`, `mariadbUser` and `mariadbPassword` should be configured on the mariadb chart, similar to how `mariadbRootPassword` is set.

**Review by sameersbn — CHANGES_REQUESTED** [review]
Aside from my comments, i tested the changes and found them to be working. nice job!

**Comment by prydonius** [comment]
Pulling this into Milestone 4 as it blocking other WordPress updates. @jbianquetti-nami you seem to have pulled in other commits on different charts, can you please fix that?

File: stable/wordpress/templates/secrets.yaml [hunk 5]
```diff
@@ -14,4 +14,4 @@ data:
   {{ else }}
   wordpress-password: {{ randAlphaNum 10 | b64enc | quote }}
   {{ end }}
-  smtp-password: {{ default "" .Values.smtpPassword | b64enc | quote }}
+  smtp-password: {{ default ""  .Values.smtpPassword | b64enc | quote }}
```
Comment (prydonius) [inline stable/wordpress/templates/secrets.yaml]: undo this change

File: stable/wordpress/README.md [hunk 6]
```diff
@@ -45,34 +45,38 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `imagePullPolicy`                    | Image pull policy                          | `IfNotPresent`                                 |
+| `wordpressUsername`                  | User of the application                    | `user`                                         |
+| `wordpressPassword`                  | Application password                       | _random 10 character long alphanumeric string_ |
+| `wordpressEmail`                     | Admin email                                | `user@example.com`                             |
+| `wordpressFirstName`                 | First name                                 | `FirstName`                                    |
+| `wordpressLastName`                  | Last name                                  | `LastName`                                     |
+| `wordpressBlogName`                  | Blog name                                  | `User's Blog!`                                 |
+| `allowEmptyPassword`                 | Allow DB blank passwords                   | `no`                                           |
```
Comment (sameersbn) [inline stable/wordpress/README.md]: The default value is `yes`

**Review by sameersbn — CHANGES_REQUESTED** [review]
Apart from my comment about the default value of `allowEmptyPassword`, the changes look good to me.

File: stable/wordpress/README.md (hunk 6 above)
Comment (jbianquetti-nami) [reply stable/wordpress/README.md]: Lost while rebasing. I will update it in a few minutes.

**Review by sameersbn — APPROVED** [review]

**Comment by prydonius** [comment]
Thanks @jbianquetti-nami!

**Comment by imphasing** [comment]
Would it be possible to get an example of how I would use helm to install wordpress while specifying a separate mysql host, not creating one on the cluster? I'm not seeing a way to skip installation of mariadb and specify a remote host, which I think is what this PR was intended for based on the title
